company/views.py

```diff
@@ -251,26 +268,148 @@ def get_context_data(self, 	**kwargs):
 		context['upc'] = UserPostConnection.objects.filter(company_id= kwargs['pk'],status="Rejected") 
 		return context
 
-def AddQuestionView(request):
-		
+class GenerateQuestionView(TemplateView):
+	template_name = 'company/test.html'
+
+	def get_context_data(self, 	**kwargs):
+		context = super().get_context_data(**kwargs)
+		context['technology'] = Technology.objects.all()
+		return context 
+
+	def post(self, request, *args, **kwargs):
+		return HttpResponseRedirect('/company/addquestion/')
+
+class ListofQuestionView(TemplateView):
+	template_name = 'company/addquestion.html'
+
+	def get_context_data(self, 	**kwargs):
+		context = super().get_context_data(**kwargs)
+		context['technology'] = Technology.objects.all()
+		tech = Technology.objects.get(technology_name = kwargs['type'])
+		context['type'] = kwargs['type']
+		context['questions'] = Question.objects.filter(technology_id = tech.pk)
+		context['answers'] = Answers_HR.objects.all()
+		return context
 
+	def post(self,request, *args, **kwargs):
+		try:
+			question = request.POST.get('question')
+			answer = request.POST.get('answer')
+			print("before IF")
+			if request.POST.get('question') != "" and request.POST.get('answer') != "":
+
+				tech = Technology.objects.get(technology_name = kwargs['type'])
+				print("TEchnology is",tech)
+				obj1 = Answers_HR()
+				obj = Question()
+				obj.text = question
+				obj.company_id = request.user.id
+				obj.technology_id = tech.pk
+				obj.save()
+
+				
+				obj1.question_id = obj.pk
+				obj1.text = answer
+				obj1.save()
+				print("obj is ---------------",obj)
+				response = JsonResponse({'obj_id': obj.id,'obj_text':obj.text,'obj_technologyid':obj.technology_id,'obj1_text':obj1.text},safe =False)
+				
+				return  response
+		except Exception as e:
+			print(e)
+
+class GenerateTestView(TemplateView):
+	template_name = "company/generateTest.html"
 
-	ChoiceFormSet = formset_factory(AnswerForm,validate_min=True)
-	CP = CompanyProfile.objects.get(user = request.user)
-	company =  CompanyProfile.objects.filter(user_id = CP.user_id)
-	if request.method == 'POST':
-		form = QuestionForm(request.POST)
-		formset = ChoiceFormSet(request.POST)
-
-		if all([form.is_valid(),formset.is_valid()]):
-			question = form.save()
-			for inline_form in formset:
-				choice = inline_form.save(commit=False)
-				choice.questions = question
-				choice.save()
-			return render(request, 'company/test.html',{})
-	else:
-		form = QuestionForm()
-		formset = ChoiceFormSet()
-
-	return render(request,'company/test.html',{ 'form': form, 'formset' : formset,'company': company })
\ No newline at end of file
+	def get_context_data(self, **kwargs):
+		context = super().get_context_data(**kwargs)
+		idd = kwargs['id']
```

Comment:
what is idd it should be id if its conflicting then it should be hr_id or application_id